tensorflow/contrib/lite/testing/lstm_test.py

```diff
@@ -0,0 +1,274 @@
+  """Convert a model's graph def into a tflite model."""
+
+  data_types = [_TF_TYPE_INFO[x[2]][1] for x in input_tensors]
+  opts = toco_options(
+      data_types=data_types,
+      input_arrays=[x[0] for x in input_tensors],
+      shapes=[x[1] for x in input_tensors],
+      output_arrays=output_tensors,
+      output_format=output_format)
+
+  with tempfile.NamedTemporaryFile() as graphdef_file, \
+       tempfile.NamedTemporaryFile() as output_file, \
+       tempfile.NamedTemporaryFile("w+") as stdout_file:
+    graphdef_file.write(graph_def_str)
+    graphdef_file.flush()
+
+    cmd = ("%s --input_file=%s --output_file=%s %s > %s " %
+           (FLAGS.toco, graphdef_file.name, output_file.name, opts,
+            stdout_file.name))
+    exit_code = os.system(cmd)
+    return (None if exit_code != 0 else output_file.read())
+
+def main(unused_args):
+  """Main"""
+
+  if not os.path.exists(FLAGS.toco):
+    raise RuntimeError("%s: not found" % FLAGS.toco)
+
+  if FLAGS.output_path:
+    if not os.path.exists(FLAGS.output_path):
+      os.makedirs(FLAGS.output_path)
+      if not os.path.isdir(FLAGS.output_path):
+        raise RuntimeError("Failed to create dir %r" % FLAGS.output_path)
+
+  for parameters in test_parameters:
+    keys = parameters.keys()
+    for curr in itertools.product(*parameters.values()):
+      param_dict = dict(zip(keys, curr))
+
+      def build_example(param_dict_real):
+        """Build the model with parameter values set in param_dict_real."""
+
+        np.random.seed(RANDOM_SEED)
+
+        # Build graph
+        tf.reset_default_graph()
+
+        with tf.device("/cpu:0"):
+          try:
+            inputs, outputs = build_graph(param_dict_real)
+          except (tf.errors.UnimplementedError, tf.errors.InvalidArgumentError,
+                  ValueError):
+            return None
+
+        sess = tf.Session()
+        sess.run(tf.global_variables_initializer())
+
+        try:
+          baseline_inputs, baseline_outputs = build_input(
+              param_dict_real, sess, inputs, outputs)
+        except (tf.errors.UnimplementedError, tf.errors.InvalidArgumentError,
+                ValueError):
+          return None
+
+        # Convert graph to toco
+        input_tensors = [(input_tensor.name.split(":")[0],
+                          input_tensor.get_shape(), input_tensor.dtype)
+                         for input_tensor in inputs]
+        output_tensors = [normalize_output_name(out.name) for out in outputs]
+
+        graph_def = freeze_graph(
+            sess,
+            tf.global_variables() + inputs + outputs) if FLAGS.use_frozen_graph else sess.graph_def
+        model_binary = toco_convert(
+            graph_def.SerializeToString(), input_tensors, output_tensors)
+
+        # Check if `LstmCell` is contained
+        print("========================")
+        if 'label=\"LstmCell\"' in model_binary.decode('utf-8'):
+          print("Found 'LstmCell'")
+        else:
```

Comment:
It's not always safe to decode the model to UTF8. 
Doesn't the generated example test cover this?